tools/nitrogfx/gfx.c

```diff
@@ -342,3 +375,66 @@ void WriteGbaPalette(char *path, struct Palette *palette)
 
 	fclose(fp);
 }
+
+void WriteNtrPalette(char *path, struct Palette *palette, bool ncpr)
+{
+    FILE *fp = fopen(path, "wb");
+
+    if (fp == NULL)
+        FATAL_ERROR("Failed too open \"%s\" for writing.\n", path);
+
+    uint32_t size = palette->numColors * 2;
+    uint32_t extSize = size + (ncpr ? 0x10 : 0x18);
+
+    //NCLR header
+    WriteGenericNtrHeader(fp, (ncpr ? "RPCN" : "RLCN"), extSize, !ncpr);
+
+    //PLTT header
+    //magic number
+    fputs("TTLP", fp);
+
+    //section size
+    fputc(extSize & 0xFF, fp);
```

Comment:
Repeated calls to fputc are inefficient. Define the structure in memory, then write it in one fwrite call.